src/khoj/processor/conversation/utils.py

```diff
@@ -186,21 +188,27 @@ def generate_chatml_messages_with_context(
     # Extract Chat History for Context
     chatml_messages: List[ChatMessage] = []
     for chat in conversation_log.get("chat", []):
-        message_notes = f'\n\n Notes:\n{chat.get("context")}' if chat.get("context") else "\n"
-        role = "user" if chat["by"] == "you" else "assistant"
-
-        if chat["by"] == "khoj" and "excalidraw" in chat["intent"].get("type"):
-            message_content = chat.get("intent").get("inferred-queries")[0] + message_notes
-        else:
-            message_content = chat["message"] + message_notes
+        message_context = ""
+        if chat["by"] == "khoj" and "excalidraw" in chat["intent"].get("type", ""):
+            message_context += chat.get("intent").get("inferred-queries")[0]
+        if not is_none_or_empty(chat.get("context")):
+            references = "\n\n".join(
+                {f"# File: {item['file']}\n## {item['compiled']}\n" for item in chat.get("context") or []}
```

Comment:
`compiled` helps us with retrieving the correct embeddings during search. Should we use `raw` for injecting context in the query?